contracts/libraries/ErrorMessage.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+
+pragma solidity >=0.7.0 <0.9.0;
+
+/**
+ * @title Error Message - Contract which uses assembly to revert with a custom error message.
+ * @author Shebin John - @remedcu
+ * @notice The aim is to save gas using assembly to revert with custom error message.
+ */
+abstract contract ErrorMessage {
```

Comment:
Personally, I'm fine with this assembly, it is fairly simple, and removing all the additional string encoding saves non-trivial amount of space.

We should probably have a discussion with @rmeissner about what our appetite for assembly in the contracts is.

Reply:
I think the benefits of the approach outweigh the cognitive overhead of assembly code